sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java

```diff
@@ -421,25 +424,30 @@
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.storage.blob.specialized.PageBlobAsyncClient.getPageRangesDiffWithResponse#BlobRange-String-BlobAccessConditions}
+     * {@codesnippet com.azure.storage.blob.specialized.PageBlobAsyncClient.getPageRangesDiffWithResponse#PageRange-String-BlobAccessConditions}
      *
-     * @param blobRange {@link BlobRange}
+     * @param pageRange {@link PageRange}
      * @param prevSnapshot Specifies that the response will contain only pages that were changed between target blob and
      * previous snapshot. Changed pages include both updated and cleared pages. The target blob may be a snapshot, as
      * long as the snapshot specified by prevsnapshot is the older of the two.
      * @param accessConditions {@link BlobAccessConditions}
      * @return A reactive response emitting all the different page ranges.
      * @throws IllegalArgumentException If {@code prevSnapshot} is {@code null}
      */
-    public Mono<Response<PageList>> getPageRangesDiffWithResponse(BlobRange blobRange, String prevSnapshot,
+    public Mono<Response<PageList>> getPageRangesDiffWithResponse(PageRange pageRange, String prevSnapshot,
         BlobAccessConditions accessConditions) {
         return withContext(context ->
-            getPageRangesDiffWithResponse(blobRange, prevSnapshot, accessConditions, context));
+            getPageRangesDiffWithResponse(pageRange, prevSnapshot, accessConditions, context));
     }
 
-    Mono<Response<PageList>> getPageRangesDiffWithResponse(BlobRange blobRange, String prevSnapshot,
+    Mono<Response<PageList>> getPageRangesDiffWithResponse(PageRange pageRange, String prevSnapshot,
         BlobAccessConditions accessConditions, Context context) {
-        blobRange = blobRange == null ? new BlobRange(0) : blobRange;
+        if (pageRange == null) {
+            // Throwing is preferred to Single.error because this will error out immediately instead of waiting until
+            // subscription.
+            throw logger.logExceptionAsError(new IllegalArgumentException("pageRange cannot be null."));
+        }
```

Comment:
Same question here, if it is because there is no constructor for ’PageRange’ like ’BlobRange’ is there a possible substitute?

Reply:
PageRange required to specify the start and end. I don't think we need to set default to it.
https://docs.microsoft.com/en-us/rest/api/storageservices/put-page

Reply:
PutRange needs this check, but get pages doesn't (which this uses).

https://docs.microsoft.com/en-us/rest/api/storageservices/get-page-ranges

Reply:
Make sense.